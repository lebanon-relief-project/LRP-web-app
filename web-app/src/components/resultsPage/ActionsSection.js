import React, { useState } from "react";
import styled from "styled-components";
import devices from "../../styles/Devices";
import { Routes } from "../../App.js";
import Card from "../Card";
import PDF from "../../assets/downloads/Self-Help_Exercises.pdf";
import ReachOutModal from "./ReachOutModal";

export const ActionSection = () => {
  const [reachOutModalVisible, setReachOutModalVisible] = useState(false);
  return (
    <>
      <Wrapper>
        <Title>Some actions you can take</Title>
        <CardWrapper>
          <StyledCard
            title="Support"
            buttonText="Reach Out"
            path="#"
            onClick={() => {
              setReachOutModalVisible(true);
            }}
          >
            <StyledParagraph>
              Surround yourself with friends and family. And remember, you can
              always reach outâ€¦
            </StyledParagraph>
          </StyledCard>
          <StyledCard
            title="Useful exercises"
            buttonText="View exercises"
            externalPath={PDF}
          >
            <StyledParagraph>
              Here are some exercises to reduce stress and regain control over
              your actions and thoughts.
            </StyledParagraph>
          </StyledCard>
          <StyledCard
            title="14 days of wellbeing"
            disabled
            path={Routes.HELP}
            buttonText="Coming Soon"
          >
            <StyledParagraph>
              Keep your mind and body engaged by joining our 14 days of
              wellbeing.
            </StyledParagraph>
          </StyledCard>
        </CardWrapper>
      </Wrapper>
      {reachOutModalVisible && (
        <ReachOutModal
          closeModal={() => {
            setReachOutModalVisible(false);
          }}
        />
      )}
    </>
  );
};

const StyledParagraph = styled.p`
  min-height: 7rem;
`;

const Wrapper = styled.div`
  width: 100%;
  max-width: 960px;

  padding-top: 2rem;
  padding-bottom: 2rem;
  margin: auto;
  background: inherit;

  @media (max-width: ${devices.ipadpro}) {
    margin: 25px 18px;
    padding-top: 0px;
    padding-bottom: 0px;
    width: auto;
    min-width: 0;
  }
`;

const Title = styled.h2`
  font-family: Playfair Display;
  font-style: normal;
  font-weight: bold;
  font-size: 30px;
  line-height: 38px;
`;

const StyledCard = styled(Card)`
  height: auto;
  width: 30%;
  min-width: 160px;
  margin: 5% 0;
  legend {
    width: auto;
    margin: 0 auto;
  }
  @media (max-width: ${devices.mobile}) {
    width: 100%;
  }
`;

const CardWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  @media (max-width: ${devices.mobile}) {
    flex-direction: column;
  }
`;

export default ActionSection;
