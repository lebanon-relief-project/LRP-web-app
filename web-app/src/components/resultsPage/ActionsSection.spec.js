import React from "react";
import { render } from "@testing-library/react";
import ActionsSection from "./ActionsSection";
import "jest-styled-components";
import { fireEvent } from "@testing-library/dom";
import { act } from "react-dom/test-utils";

jest.mock("../Card", (props) => {
  return (props) => (
    <div data-testid={`${props.title}`} onClick={props.onClick}>
      {props.title}
    </div>
  );
});

jest.mock("./ReachOutModal", () => {
  return () => <div data-testid="modal">Reach Out Modal</div>;
});

describe("the ActionsSection component", () => {
  let container;

  it("should match the snapshot", () => {
    ({ container } = render(<ActionsSection />));

    expect(container).toMatchSnapshot();
  });

  it("should open Modal when Reach Out button is clicked", () => {
    let getByTestId, queryByTestId;
    ({ container, getByTestId, queryByTestId } = render(<ActionsSection />));

    expect(queryByTestId("modal")).toBeFalsy();

    act(() => {
      fireEvent.click(getByTestId("Support"));
    });

    expect(queryByTestId("modal")).toBeTruthy();
  });
});
