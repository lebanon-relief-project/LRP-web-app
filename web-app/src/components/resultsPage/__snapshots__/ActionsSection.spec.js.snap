// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ActionsSection component should match the snapshot 1`] = `
.c0 {
  width: 100%;
  max-width: 960px;
  padding-top: 2rem;
  padding-bottom: 2rem;
  margin: auto;
  background: white;
}

.c1 {
  font-family: Playfair Display;
  font-style: normal;
  font-weight: bold;
  font-size: 30px;
  line-height: 38px;
}

@media (max-width:1025px) {
  .c0 {
    margin: 25px 0px;
    padding-top: 0px;
    padding-bottom: 0px;
    width: auto;
    min-width: 0;
  }
}

@media (max-width:1025px) {
  .c1 {
    margin: 0px 18px;
  }
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      Some actions you can take
    </h2>
    <div>
      CardGroup
    </div>
  </div>
</div>
`;
